src/main/java/de/bsi/secvisogram/csaf_cms_backend/json/AdvisoryWrapper.java

```diff
@@ -754,9 +811,132 @@ public static JsonNode applyJsonPatchToNode(JsonNode patch, JsonNode source) {
+
+        entry.put("summary", summary);
+        entry.put("url", url);
+        return this;
+    }
+
+    public void setTemporaryTrackingId(String trackingidCompany, String trackingidDigits, long sequentialNumber) {
+
+        String companyName = calculateCompanyName(trackingidCompany);
+        String formatted = formatNumber(trackingidDigits, sequentialNumber);
+        setDocumentTrackingId(companyName + "-TEMP-" + formatted);
+    }
+
+    /**
+     * Set the final tracking id in the advisory and a DocumentReferencesNode with the url of the tracking id
+     * @param baseUrl the configured base url
+     * @param trackingIdCompany the configured company for the name of the tracking id
+     * @param trackingIdDigits the count of leading zeros to which the sequentialNumber is filled with
+     * @param sequentialNumber the next sequentialNumber
+     */
+    public void setFinalTrackingIdAndUrl(String baseUrl, String trackingIdCompany, String trackingIdDigits, long sequentialNumber) {
+
+        setTempTrackingIdInMeta(getDocumentTrackingId());
+
+        String companyName = calculateCompanyName(trackingIdCompany);
+        String formatted = formatNumber(trackingIdDigits, sequentialNumber);
+        int year = calculatePublishYear();
+        String trackingId = companyName + "-" + year + "-" + formatted;
+        setDocumentTrackingId(trackingId);
+
+        if (baseUrl != null && !baseUrl.isBlank()) {
+            String referenceUrl = calculateReferenceUrl(baseUrl, trackingId);
+            this.addDocumentReferencesNode("URL generated by system", referenceUrl);
+        }
+    }
+
+    /**
+     * Generate the url for the given baseUrl and tracking id.
+     * Format: BaseURL/TLPLabel/YearOfPublication/trackingId.json
+     * @param baseUrl configured base url
+     * @param trackingId the trackingid
+     * @return the calculated url
+     */
+    String calculateReferenceUrl(String baseUrl, String trackingId) {
+
+        int year = calculatePublishYear();
+        String fileName = calculateFileName(trackingId);
+        String tlpLabel = getDocumentDistributionTlp() != null ? getDocumentDistributionTlp() : "WHITE";
+        return baseUrl + "/" + tlpLabel + "/" + year + "/" + fileName;
     }
 
+    /**
+     * If trackingidCompany is  not set then we generate the abbreviation from the Publisher Name.
+     * Always the first letters of the words separated by spaces
+     * @param trackingidCompany configured company short name
+     * @return the calculated name
+     */
+    String calculateCompanyName(String trackingidCompany) {
+        String companyName = trackingidCompany;
+        if (trackingidCompany == null || trackingidCompany.isBlank()) {
+            String publisherName = this.getDocumentPublisherName();
+            publisherName = (publisherName != null) ? publisherName.trim() : "";
+            companyName = (publisherName.indexOf(' ') > 0) ? publisherName.substring(0, publisherName.indexOf(' ')) : "";
```

Comment:
Even if there is only one word in `publisherName` we need to use the first letter instead of an empty string.

Reply:
fixed